contracts/test/DelegateE2E.t.sol

```diff
@@ -369,6 +376,133 @@ contract TestIntegrationJBXBuybackDelegate is Test, UniswapV3ForgeQuoter {
     assertEq(jbController.reservedTokenBalanceOf(1), _reservedBalanceBefore);
   }
 
+    /**
+   * @notice If the amount of token returned by swapping is greater than by minting, swap & use quote from uniswap lib
+   *
+   * @dev    Should swap for both beneficiary and reserve (by burning/minting)
+   */
+  function test_swapWhenQuoteNotProvidedInMetadata(uint256 _amountIn) public {
+    _amountIn = bound(_amountIn, 100, 100 ether);
+  
+    // Reconfigure with a weight of 1  
+    _reconfigure(1, address(delegate), 1, 0);
+
+    uint256 _reservedBalanceBefore = jbController.reservedTokenBalanceOf(1);
+
+    uint256 _quote = getAmountOut(pool, _amountIn, address(weth));
+    
+    vm.expectEmit(true, true, true, true);
+    emit JBXBuybackDelegate_Swap(1, _amountIn, _quote);
+
+    // Pay the project
+    jbEthPaymentTerminal.pay{value: _amountIn}(
+      1,
+      _amountIn,
+      address(0),
+      address(123),
+      /* _minReturnedTokens */
+      0,
+      /* _preferClaimedTokens */
+      true,
+      /* _memo */
+      'Take my money!',
+      /* _delegateMetadata */
+      new bytes(0)
+    );
+
+    // Check: token received by the beneficiary
+    assertEq(jbx.balanceOf(address(123)), _quote);
+
+    // Check: reserve unchanged
+    assertEq(jbController.reservedTokenBalanceOf(1), _reservedBalanceBefore);
+  }
+
+  /**
+   * @notice If the amount of token returned by swapping is greater than by minting, swap & use quote from uniswap lib when cardinality is increased
+   *
+   * @dev    Should swap for both beneficiary and reserve (by burning/minting)
+   */
+  function test_swapWhenCardinalityIsIncreased(uint256 _amountIn) public {
+    _amountIn = bound(_amountIn, 100, 100 ether);
+  
+    // Reconfigure with a weight of 1  
+    _reconfigure(1, address(delegate), 1, 0);
+
+    uint256 _reservedBalanceBefore = jbController.reservedTokenBalanceOf(1);
+
+    uint256 _quote = getAmountOut(pool, _amountIn, address(weth));
+
+    delegate.increaseSecondsAgo(200000);
```

Comment:
same, the actual twap over that period should be tested (greather than the weight)

Reply:
already doing seems can be resolved no ?